R/spa.R

```diff
@@ -0,0 +1,83 @@
+### new SPA function
+
+SPA_pval <- function(score.result, nullmod, G, pval.thresh = 0.05){
+	if (!requireNamespace("SPAtest")) stop("package 'SPAtest' must be installed to calculate SPA p-values")
+	expit <- function(x){exp(x)/(1+exp(x))}
+
+	# add columns to score.results
+	score.result$SPA.pval <- NA_real_
+	score.result$SPA.converged <- NA_real_
+
+	# index for variants with Score.pval < pval.thresh
+	# only bother running SPA on these variants
+	idx <- which(score.result$Score.pval <= pval.thresh)
```

Comment:
@smgogarten I took this idea from the original saige code. Basically, it only does the SPA adjustment for variants with score pval < pval.thresh. I think this makes sense from a speed perspective. However, it does lead to the `SPA.pval` output in the results having `NA` values for the variants not tested. That may not be ideal for post processing qq plots and such. 

We could just copy `Score.pval` over into `SPA.pval` for those variants not run, but perhaps that's misleading. What do you think?

Reply:
I would copy Score.pval into SPA.pval, because everyone will want to look at all the p-values together. We should just make clear in the documentation that it is happening.

Reply:
Should we do that or overwrite the `Score.pval` with the `SPA.pval`? This ties into my question above as well. If we overwrote, then the `SPA.converged` flag would actually be an indicator of if SPA ran and converged (`TRUE`), ran but didn't converge ('FALSE'), or wasn't run ('NA'). We could only overwrite if converged == `TRUE`

Reply:
Would anyone ever want both p-values to compare them? If no, then overwriting makes sense. We could also check what SMMAT returns, for consistency - I think it gives multiple p-values. Can you do SPA for a burden test, and if so do we need to add this argument to assocTestAggregate as well?

Reply:
Would anyone want both? probably not in production; possibly for testing. I think I'm leaning toward overwriting now. The BR and CMP code overwrite the score p-values when that option is selected. 

SMMAT returns multiple p-values: the burden p-value, the "SKAT" like p-values, and then the combined SMMAT p-value. This makes more sense here because they are all p-values for different tests. 

I guess I don't see any reason why you couldn't apply SPA to burden; the genotype vector would just be replaced by the burden vector. They seem to have done a similar thing with BR and CMP. I guess I should add this here as well.

Reply:
Tamar says that SPA doesn't apply to the burden test: "The p-value computation under SPA uses the allele frequency. This is irrelevant in burden test. We originally tried this as well, but realized it’s inappropriate and while sometimes it looks okay sometimes it give bad results (because it’s not suppose to work)."